### PR TITLE
EVG-17370: add option to apply Amboy queue settings for queues matching a regexp

### PR DESCRIPTION
**Pull request by Kimchelly**
Jira: https://jira.mongodb.org/browse/EVG-17370

### Description 
Add admin settings to apply Amboy queue settings to named queues using a regexp instead of exact name match.

### Testing 
Added admin settings unit tests and tested the settings in local Evergreen.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -128,8 +128,9 @@ github.com/aws/aws-sdk-go v1.15.11/go.mod h1:mFuSZ37Z9YOHbQEwBWztmVzqXrEkub65tZo
 github.com/aws/aws-sdk-go v1.41.11/go.mod h1:585smgzpB/KqRA+K3y/NL/oYRqQvpNJYvLm+LY1U59Q=
 github.com/aws/aws-sdk-go v1.41.12/go.mod h1:585smgzpB/KqRA+K3y/NL/oYRqQvpNJYvLm+LY1U59Q=
 github.com/aws/aws-sdk-go v1.43.30/go.mod h1:y4AeaBuwd2Lk+GepC1E9v0qOiTws0MIWAX4oIKwKHZo=
-github.com/aws/aws-sdk-go v1.44.46 h1:BsKENvu24eXg7CWQ2wJAjKbDFkGP+hBtxKJIR3UdcB8=
```
Comment (malikchaya2) [inline go.sum:131]: are these changes intentional? I'm not used to seeing them in PRs

Comment (Kimchelly) [reply go.sum:131]: Yes, this is standard behavior for go module upgrades and tidying. I had to upgrade Amboy to include the necessary features from https://github.com/mongodb/amboy/pull/287.

**Review by malikchaya2 — APPROVED** [review]
